### PR TITLE
Added a new entry for Decision Trees

### LINKED ISSUE #4931: [Concept Entry] Sklearn decision-trees
### Reason/inspiration (optional)

We would like a new entry on the `Decision Trees` concept under Sklearn. The entry should go in a new file under `docs/content/sklearn/concepts/decision-trees/decision-trees.md`.

The entry should include:

- An introduction to the concept
- A `Syntax` section that provides the syntax for the concept (if applicable)
- An `Example` section that provides an example demonstrating the concept in use (if applicable)
- A `Codebyte Example` section that provides a codebyte example demonstrating the concept in use (if applicable)

Please refer to the [concept entry template](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entrty for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by mamtawardhani**
### Description
Added a new entry for Decision Trees concept

### Issue Solved
Closes #4931 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [ ] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/sklearn/concepts/decision-trees/decision-trees.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  - 'Machine Learning'
+Tags:
+  - 'Machine Learning'
+  - 'Scikit-learn'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+---
+
+**Decision Trees** are intuitive machine learning algorithms that can be used for classification and regression tasks. They work by recursively splitting data into subsets based on feature values, building a tree-like structure of decisions to predict an output. Decision trees are widely favored for their ease of use and clarity in interpretation.
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/decision-trees/decision-trees.md:15]: The second sentence is slightly wordy and could be more concise like this -

```suggestion
**Decision Trees** are intuitive machine learning algorithms that can be used for classification and regression tasks. They recursively split data into subsets based on feature values, creating a tree-like structure to make predictions. 

Decision trees are widely favored for their ease of use and clarity in interpretation.
```

File: content/sklearn/concepts/decision-trees/decision-trees.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+In [sklearn](https://www.codecademy.com/resources/docs/sklearn), decision trees are implemented in the `DecisionTreeClassifier` and `DecisionTreeRegressor` classes. They can handle both numerical and categorical data, and support advanced features like pruning and handling missing values.
+
+## Advantages of Decision Trees
+
+- Easy to understand and interpret.
+- Handles both numerical and categorical data.
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/decision-trees/decision-trees.md:21]: A verb in third person might better match the other bullet points.

```suggestion
- Can handle both numerical and categorical data.
```

File: content/sklearn/concepts/decision-trees/decision-trees.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+- Requires minimal data preprocessing (e.g., no need for feature scaling).
+- Can model non-linear relationships.
+
+## Limitations
+
+- Prone to overfitting without proper pruning.
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/decision-trees/decision-trees.md:27]: A verb in the third person might better match the other bullet points.

```suggestion
- Can be prone to overfitting without proper pruning.
```

File: content/sklearn/concepts/decision-trees/decision-trees.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+- Sensitive to small changes in the data, which can lead to different tree structures.
+
+## Syntax
+
+Below is an example syntax of decision tree classifier:
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/decision-trees/decision-trees.md:32]: This can be rewritten for more clarity and readability.

```suggestion
Below is an example of the syntax for using a decision tree classifier:
```

File: content/sklearn/concepts/decision-trees/decision-trees.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+```pseudo
+from sklearn.tree import DecisionTreeClassifier
+
+# Initializing the Decision Tree Classifier
+model = DecisionTreeClassifier()
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/decision-trees/decision-trees.md:38]: We should add/explain commonly used parameters such as `max_depth`, `random_state` etc. Refer [documentation](https://scikit-learn.org/1.5/modules/generated/sklearn.tree.DecisionTreeClassifier.html) for reference.

**Review by PragatiVerma18 — CHANGES_REQUESTED** [review]
Hey @mamtawardhani, I have some suggestions, please check the comments and do the required changes. Let me know if you have any questions.

**Comment by mamtawardhani** [comment]
Hey @PragatiVerma18 the changes are done

**Review by PragatiVerma18 — APPROVED** [review]

**Comment by PragatiVerma18** [comment]
Hey, @mamtawardhani, thanks for making the changes so quickly. This looks good to go! 🚀
